great_expectations/data_context/data_context/ephemeral_data_context.py

```diff
@@ -51,6 +52,26 @@ def _init_variables(self) -> EphemeralDataContextVariables:
         )
         return variables
 
+    def _init_datasource_store(self) -> None:
+        from great_expectations.data_context.store.datasource_store import (
+            DatasourceStore,
+        )
+
+        store_name: str = "datasource_store"  # Never explicitly referenced but adheres
+        # to the convention set by other internal Stores
+        store_backend: dict = {"class_name": "InMemoryStoreBackend"}
+
+        datasource_store: DatasourceStore = DatasourceStore(
+            store_name=store_name,
+            store_backend=store_backend,
+        )
+        self._datasource_store = datasource_store
+
+        for datasource_name, datasource_config in self.config.datasources.items():
+            self._datasource_store.set_by_name(
+                datasource_name=datasource_name, datasource_config=datasource_config
+            )
```

Comment:
If we don't populate our `DatasourceStore` here, we never have the opportunity to do so. The subsequent call to `_init_datasources` iterates through `self._datasource_store.list_keys()` but that isn't particularly relevant to a store backed by an `InMemoryStoreBackend`.

I could also overwrite the `_init_datasources` method but this does the job. If we're cool with it, I can add a comment explaining the rationale.

Reply:
Not sure I fully understand but please add a comment and I'm good with that.

Reply:
This one is probably easier to discuss live if you wanted more context. As a quick summary, the emphemeral context has no way of populating it's cache unless we explicitly add some logic here.